### PR TITLE
No cache

### PR DESCRIPTION
**Pull request by bamader**
## Background
This PR _SLAUGHTERS_ all traces of the roster context as well as SWR from the client side of the app, replacing them instead with simple `useEffect(() => {}, [])` and `apiGet` calls.

## GitHub Issue
https://github.com/ctoec/data-collection/issues/1459

## Associated PRs
None

## Validation Plan
The cache touches several points in the app: the roster and all affiliated actions/flows from there (delete, withdraw, edit record, create record, batch edit). Also affected are funding spaces and reporting periods (because these had cached values stored; the funding types and times in the case of the former, and the periods themselves in the case of the latter); the file upload flow (the preview screen invoked the missing info part of an uploaded file); and the admin overview screen (which cached the summary response). My validations consisted of testing that each of these core functions could still be performed and that the numbers/info in them was accurate, including persistent and updating to reflect the result of some action (i.e. number of enrolled children reported on the roster went down by one after a delete, plus that child no longer appears; etc.).

## Automated Testing
- [x] All unit tests are passing.

## Additional Context
There _is_ still a small trace of useSWR in the old roster context that I didn't delete yet, because I didn't want to go through the labor of patching up now obsolete code just so we can delete it with this EPIC. So I left TODO's in those files to delete them and the package itself later, when we're ready to merge this into prod.

### CODE REVIEW

**Review by vonbearshark — APPROVED** [review]
🙌🏻 

couple more `swr` usages than I thought, but still--looks good!

**Review by JordanGuinn — APPROVED** [review]
WOOT

**Review by lukenewman — APPROVED** [review]
lovely

File: client/src/components/Forms/Enrollment/Funding/Fields/ContractSpace.tsx [hunk 1]
```diff
@@ -45,9 +46,16 @@ export const ContractSpaceField = <
...
   fundingAccessor = (data) => data as TObjectDriller<Funding>,
 }: ContractSpaceProps<T>) => {
-  const { data: fundingSpaces } = useAuthenticatedSWR<FundingSpace[]>(
-    `funding-spaces?${stringify({ organizationId })}`
-  );
+
+  const { accessToken } = useContext(AuthenticationContext);
+  const [fundingSpaces, setFundingSpaces] = useState<FundingSpace[]>([]);
+  useEffect(() => {
+    (async function fetchReportingPeriods() {
```
Comment (JordanGuinn) [inline client/src/components/Forms/Enrollment/Funding/Fields/ContractSpace.tsx:53]: nit: `fetchFundingSpaces`?

File: client/src/components/Forms/Enrollment/Fields/NewFunding.tsx [hunk 2]
```diff
@@ -52,9 +53,16 @@ export const NewFundingField = <
...
   hideStatus,
 }: FundingFieldProps<T>) => {
-  const { data: fundingSpaces } = useAuthenticatedSWR<FundingSpace[]>(
-    `funding-spaces?${stringify({ organizationId })}`
-  );
+
+  const { accessToken } = useContext(AuthenticationContext);
+  const [fundingSpaces, setFundingSpaces] = useState<FundingSpace[]>([]);
+  useEffect(() => {
+    (async function fetchReportingPeriods() {
```
Comment (JordanGuinn) [inline client/src/components/Forms/Enrollment/Fields/NewFunding.tsx:60]: nit: `fetchFundingSpaces`?

**Review by JordanGuinn — COMMENTED** [review]
Whoops - few minor things I nearly missed first

File: client/src/components/Forms/Enrollment/Fields/NewFunding.tsx (hunk 2 above)
Comment (bamader) [reply client/src/components/Forms/Enrollment/Fields/NewFunding.tsx:60]: LOL no I wasn't copy/pasting code, why do you ask 🤦

**Comment by JordanGuinn** [comment]
## Validation Plan
- [ ] Confirm roster still shows organization enrollments as expected
- [ ] Confirm admin overview still displays correct OEC child/site/org counts
- [ ] Confirm I can still add new fundings to enrollments without issue
- [ ] Confirm said fundings are reflected in said enrollment on save
- [ ] Confirm I can still update funding spaces for enrollment fundings without issue
- [ ] Confirm said funding spaces are reflected in said enrollment funding on save
- [ ] Confirm I can still update reporting periods for enrollment fundings without issue
- [ ] Confirm said reporting periods are reflected in said enrollment funding on save
- [ ] Confirm batch edit still functions without issue
- [ ] Confirm batch edit changes are reflected in roster view
- [ ] Confirm I can still create a child row without issue
- [ ] Confirm new child row is added to roster view
- [ ] Confirm I can still delete a child row without issue
- [ ] Confirm deleted child is removed from roster view
- [ ] Confirm I can still edit a child row without issue
- [ ] Confirm child edits are reflected in roster view
- [ ] Confirm I can still withdraw a child row without issue
- [ ] Confirm withdrawn children are reflected in roster view
- [ ] Confirm file upload still functions as expected
- [ ] Confirm uploaded children are shown in roster view

**Comment by JordanGuinn** [comment]
@bamader This all looks good to me!  I found a couple issues that I'll enumerate below, but I think these are moreso specific to the roster epic branch itself than this PR.  As such, I'd say this is ready for merge.

### Roster Epic Bugs
- The child count and age group breakdown counts at the top of the roster are inaccurate:
   - I uploaded 20 child entries my roster, but both counts inaccurately indicate that there are 28 total children.  Looks like this is actually referring to the number of _enrollments_, not the number of children.
 - There is no loading UX when navigating to the roster view on action completion anymore.  You're taken to the roster, and for a brief moment or two, it looks like you have no data before it's loaded

**Comment by bamader** [comment]
I literally had both of these exact same thoughts yesterday while working on killing the cache. I was like damn this is awkward it looks like I have no data because there's no loading wheel. + a lot for putting that in a ticket as part of the redesign (on  _all_ roster pages--edit record, create, etc.). For the enrollment count, I did notice that, but  I agree we could change some language to make that more explicit.

**Pull request merged**